packages/react-dom/src/client/ReactDOMFiberInput.js

```diff
@@ -242,7 +242,7 @@ export function postMountWrapper(
   if (name !== '') {
     node.name = '';
   }
-  node.defaultChecked = !node.defaultChecked;
+  node.defaultChecked = !node._wrapperState.initialChecked;
   node.defaultChecked = !node.defaultChecked;
```

Comment:
Remind me again why we set `defaultChecked` twice, is it related to value detachment? It'd be great if we could add a comment here explaining the reasoning.

Reply:
@aweary Actually I don't know either, but I aired on the side of avoiding speculation, haha. I can do some testing to see if we really need this. 

A lot of the value detachment stuff was added when defaultValue was assigned as an attribute through DOMPropertyOperations. I think those edge cases went away when we moved everything into ReactDOMInput. 

The thing to figure out is if assignment detaches, or if the value truly must be different. It would be easy to run a few tests.

Reply:
I think we set ```defaultChecked``` twice for solving [Checkbox `checked` state is erroneously influenced by value of `defaultChecked`](https://bugs.chromium.org/p/chromium/issues/detail?id=608416) in Chrome.

test:
```
+<!DOCTYPE html>
+<html>
+<head>
+<script src="../../../resources/testharness.js"></script>
+<script src="../../../resources/testharnessreport.js"></script>
+</head>
+<body>
+<script>
+test (function() {
+    var el = document.createElement("input");
+    el.setAttribute("type", "checkbox");
+    el.defaultChecked = true;
+    el.checked = true;
+    el.defaultChecked = false;
+    assert_true(el.checked);
+}, "This test check that checked state of checkbox once explicitly set is not affected by default state.");
+</script>
+</body>
+</html>
```
diff:
```
void HTMLInputElement::setChecked(bool nowChecked, TextFieldEventBehavior eventBehavior)
 {
+    m_reflectsCheckedAttribute = false;
     if (checked() == nowChecked)
         return;
 
-    m_reflectsCheckedAttribute = false;
     m_isChecked = nowChecked;
 
     if (RadioButtonGroupScope* scope = radioButtonGroupScope())
```

The purpose of the first assignment is set the ```m_reflectsCheckedAttribute ``` to ```false```. So, both of our codes work.

Reply:
That is a good enough answer for me, 👍

Reply:
Still, just to keep the boolean flipping in check, is it possible to do this:

```
node.defaultChecked = !node.defaultChecked;
node.defaultChecked = node._wrapperState.initialChecked;
```

?

Reply:
@nhunzaker updated.

And i changed test case because of which i found: 
if a ```<input />``` is neither checkbox nor radio, ```input.defaultChecked = true``` will call ```input.setAttribute('checked', '')``` but ```input.getAttribute('checked')``` still be ```null```. Thereafter, ```input.defaultChecked = false``` will not call ```input.setAttribute``` on the uncheckable ```input```.